packages/common/core-interfaces/src/jsonDeserialized.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { InternalUtilityTypes } from "./exposedUtilityTypes.js";
+
+/**
+ * Options for {@link JsonDeserialized}.
+ *
+ * @beta
+ */
+export interface JsonDeserializedOptions {
+	/**
+	 * A type that is managed by custom deserialization logic (beyond JSON.parse).
+	 *
+	 * The default value is `never`.
+	 */
+	Replaced: unknown;
```

Comment:
This is a bit confusing. This seems like a runtime set of options (based on it not being a generic type to take in the parameter), but never is a compile time only thing, and this value isn't optional so how does it have a default anyway?

Reply:
Breaking type out like this is the guidance for our docs. It is only to be used with JsonDeserialized type param and that is where the default exists. Looked more reasonable on the doc site.